Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/DocumentPropertyEditor/PrefabComponentAdapter.cpp

```diff
@@ -91,4 +111,158 @@ namespace AzToolsFramework::Prefab
+            AZ::Dom::Path serializedPath = propertyChangeInfo.path / AZ::Reflection::DescriptorAttributes::SerializedPath;
+
+            AZ::Dom::Path relativePathFromOwningPrefab(PrefabDomUtils::EntitiesName);
+            relativePathFromOwningPrefab /= m_entityAlias;
+            relativePathFromOwningPrefab /= PrefabDomUtils::ComponentsName;
+            relativePathFromOwningPrefab /= m_componentAlias;
+
+            AZ::Dom::Value serializedPathValue = GetContents()[serializedPath];
+            AZ_Assert(serializedPathValue.IsString(), "PrefabComponentAdapter::UpdateDomContents - SerialziedPath attribute value is not a string.");
+
+            relativePathFromOwningPrefab /= AZ::Dom::Path(serializedPathValue.GetString());
+
+
+            auto prefabFocusPublicInterface = AZ::Interface<PrefabFocusPublicInterface>::Get();
+            if (prefabFocusPublicInterface->IsOwningPrefabBeingFocused(m_entityId))
+            {
+                if (CreateAndApplyComponentEditPatch(relativePathFromOwningPrefab.ToString(), propertyChangeInfo))
+                {
+                    NotifyContentsChanged(
+                        { AZ::Dom::PatchOperation::ReplaceOperation(propertyChangeInfo.path / "Value", propertyChangeInfo.newValue) });
+
+                }
+            }
+            else if (prefabFocusPublicInterface->IsOwningPrefabInFocusHierarchy(m_entityId))
+            {
+                if (CreateAndApplyComponentOverridePatch(relativePathFromOwningPrefab, propertyChangeInfo))
+                {
+                    AZ::Dom::Patch patches(
+                        { AZ::Dom::PatchOperation::ReplaceOperation(propertyChangeInfo.path / "Value", propertyChangeInfo.newValue) });
+
+                    AZ::Dom::Path pathToProperty = propertyChangeInfo.path;
+
+                    // Get the path to parent row and its value.
+                    pathToProperty.Pop();
+                    AZ::Dom::Value propertyRowValue = GetContents()[pathToProperty];
+
+                    AZ_Assert(
+                        propertyRowValue.IsNode() &&
+                            propertyRowValue.GetNodeName().GetStringView() == AZ::DocumentPropertyEditor::Nodes::Row::Name,
+                        "PrefabComponentAdapter::UpdateDomContents - Parent path to property doesn't map to a 'Row' node. ");
+
+                    // Patch the first child in the row, which is going to the PrefabOverrideLabel.
+                    patches.PushBack(AZ::Dom::PatchOperation::ReplaceOperation(
+                        pathToProperty / 0 / PrefabPropertyEditorNodes::PrefabOverrideLabel::IsOverridden.GetName(), AZ::Dom::Value(true)));
+                    NotifyContentsChanged(patches);
+                }
+            }
+        }
+    }
+
+    bool PrefabComponentAdapter::CreateAndApplyComponentEditPatch(
+        AZStd::string_view relativePathFromOwningPrefab,
+        const AZ::DocumentPropertyEditor::ReflectionAdapter::PropertyChangeInfo& propertyChangeInfo)
```

Comment:
can remove AZ::DocumentPropertyEditor::ReflectionAdapter.
same for below CreateAndApplyComponentOverridePatch